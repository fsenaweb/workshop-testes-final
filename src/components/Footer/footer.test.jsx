import { render, screen } from "@testing-library/react";

import Footer from ".";

describe("<Footer />", () => {
  it("Verificar se o componente Footer é renderizado", () => {
    const { container } = render(<Footer />);

    const yearText = screen.getByText(
      `© ${new Date().getFullYear()} Matheus Ricelly`
    );

    expect(yearText).toBeInTheDocument();

    const link = screen.getByTestId("link");
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("href", "http://fsenaweb.dev/");
    expect(link).toHaveTextContent("fsenaweb.dev");

    expect(container).toMatchSnapshot();
  });

});
