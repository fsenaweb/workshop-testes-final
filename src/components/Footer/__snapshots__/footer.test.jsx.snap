// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Verificar se o componente Footer é renderizado 1`] = `
<div>
  <footer
    class="text-center text-body-secondary py-3 my-4 border-top"
  >
    <div
      class="text-center text-body-secondary"
    >
      <span>
        © 
        2023
         Matheus Ricelly
      </span>
      <div>
        <a
          class="text-black text-decoration-none"
          data-testid="link"
          href="http://fsenaweb.dev/"
          rel="noopener noreferrer"
          target="_blank"
        >
          fsenaweb.dev
        </a>
      </div>
    </div>
  </footer>
</div>
`;
